### ISSUE TITLE
Collection elements injection scope

### ISSUE THREAD

**Issue by stariy95** [bootique/bootique-di#19]
Right now there is no way to control collection element scope. 
This can lead to unexpected results.
For example we have an injector with singleton scope by default and want to bind some Map _unscoped_.
We can try something like this `binder.bindeMap(..).withoutScope()`. This indeed will return new map instance on each provider call, however content will be always the same as all elements inside map will be bound in default scope (i.e. singleton).

We have two options here:
* use map scope on all elements (except for elements bound by keys, as we can't control their bindings)
* allow to define scope for every element in the collection

**Comment by andrus** [comment]
Any problem with keeping Collection and its elements scopes independent from each other?

**Comment by stariy95** [comment]
@andrus yes it have its own scope right now, you can't control it effectively (or am I missing something?) :)

**Comment by andrus** [comment]
Ok, I may have misunderstood what is proposed here. So you are suggesting adding scope to collections themselves, and then make a distinction between the scopes of collection elements bound implicitly and explicitly?

**Comment by stariy95** [comment]
Yes, something like this. But I'm not sure it is a feature we need right now, I've just found this confusing as a user.
